plugins/plugin-optimize/lib/html.js

```diff
@@ -0,0 +1,115 @@
+/**
+ * Logic for optimizing .html files (note: this will )
+ */
+const fs = require('fs');
+const path = require('path');
+const hypertag = require('hypertag');
+const {injectHTML} = require('node-inject-html');
+const {projectURL, insert} = require('../util');
+const {scanJS} = require('./js');
+
+/** Scan HTML for static imports */
+async function scanHTML(htmlFiles, buildDirectory) {
+  const importList = {};
+  await Promise.all(
+    htmlFiles.map(async (file) => {
+      // TODO: add debug in plugins?
+      // log(`scanning ${projectURL(file, buildDirectory)} for imports`, 'debug');
+
+      const allCSSImports = new Set(); // all CSS imports for this HTML file
+      const allJSImports = new Set(); // all JS imports for this HTML file
+      const entry = new Set(); // keep track of HTML entry files
+
+      const code = await fs.promises.readFile(file, 'utf-8');
+
+      // <link>
+      hypertag(code, 'link').forEach((link) => {
```

Comment:
❤️ Are you kidding me? Can’t believe we weren’t using this before!

Reply:
Nice! I'm a bit concerned about the 0.0.3 version number, but the API and performance of hypertag sure looks great

Reply:
Yeah, I feel you. I took a look at the source, and it’s basically just really, really good RegEx-ing that was doing a better job than ours. It’s a simple, 0-dep library that I feel confident in its simplicity. But if we discovered a critical bug and the author didn’t respond, it’s also simple enough to fork & patch if need be.

Reply:
Yea, agreed on the feeling of this not being a super well used library that hasn't seen an update in 1+ years. BUT as long as you've given it a look inside and it looks good, then I'm +1